tests/unit/utils/test_misc.py

```diff
@@ -40,13 +41,16 @@ def test_find_application_ok_c(self):
     def test_find_application_ok_rust(self):
         device, sdk, appname = "device", "rust", "rustapp"
         with temporary_directory() as dir_path:
-            tmp_dir = (dir_path / "target" / device / "release")
+            os.chdir(dir_path)
+            cmd = ["cargo", "new", appname]
+            subprocess.check_output(cmd)
+            app_path = dir_path / appname
+            os.chdir(app_path)
```

Comment:
Is it necessary to `chdir` here? All path should be absolute.

Reply:
Yes correct

Reply:
Well actually it is mandatory for the test to pass as we must be in the Rust app root to use `cargo metadata` cmd.

Reply:
In that case it should rather be the Ragger code `output = subprocess.check_output(["cargo", "metadata", "--no-deps"])` which should explicit the `cwd` where it will be executed (`base_dir` I think?), rather than the whole system being in the correct directory when executing it; otherwise this piece of code would not be very robust.

Reply:
Yes correct